lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -48,10 +48,28 @@ Value createConstantI32(Location loc, PatternRewriter &rewriter, int32_t v) {
                                            IntegerAttr::get(i32ty, v));
 }
 
+// Create a index type constant.
+Value createIndexConstant(OpBuilder &builder, Location loc,
+
+                          TypeConverter *converter, int64_t value) {
+  Type ty = converter->convertType(builder.getIndexType());
+  return builder.create<LLVM::ConstantOp>(loc, ty,
+                                          builder.getIntegerAttr(ty, value));
+}
+
+// Create an integer constant of \param width bits.
+Value createLLVMIntegerConstant(OpBuilder &builder, Location loc, int width,
+                                int64_t value) {
```

Comment:
Do we need to check whether the `width` is wide enough to hold the `value`?

Reply:
I think that might be unnecessary, since we pass `width` and `value` directly to LLVM IR, 

```c++
Type ty = builder.getIntegerType(width);
  return builder.create<LLVM::ConstantOp>(loc, ty,
                                          builder.getIntegerAttr(ty, value));
```

We'd better to leave the underlying LLVM IR to determine whether the `width` is compatible with `value or just overflow.